src/pages/PostPage/PostSearchPage/PostSearchPage.jsx

```diff
@@ -1,3 +1,57 @@
+import styles from './PostSearchPage.module.css';
+import React, { useState, useEffect } from 'react';
+import { useLocation } from 'react-router-dom';
+import { Search } from '../../../components/Search';
+import { PLACEHOLDER } from '../../../constants';
+import { BackAppBar } from '../../../components/AppBar';
+import { PostBar } from '../../../components/PostBar';
+import { POST_LIST } from '../../../dummy/data/postList.js';
+
 export default function PostSearchPage() {
-  return <div>PostSearchPage</div>;
+  const { pathname } = useLocation();
+  const current = pathname.split('/')[2];
+  const [keyword, setKeyword] = useState('url에서 가져온 query값');
+  const [noResult, setNoResult] = useState(false);
+
+  const handleSearch = (text) => {
+    setKeyword(text);
+  };
+
+  useEffect(() => {
+    if (keyword !== '') {
+      console.log(keyword);
+    }
+  }, [keyword]);
+
+  return (
+    <div className={styles.container}>
+      <BackAppBar
+        children={
+          <Search
+            placeholder={PLACEHOLDER[current]}
+            onSearch={handleSearch}
+            keyWord={keyword}
+          />
+        }
+        hasSearchInput={true}
+      />
+      <div className={styles.content}>
+        {keyword === '' ? null : (
+          <>
+            {noResult ? (
+              <div className={styles.noResult}>검색 결과가 없습니다</div>
+            ) : (
+              <div className={styles.posts}>
+                {POST_LIST
+                  ? POST_LIST.map((post) => (
+                      <PostBar key={post.postId} data={post} />
+                    ))
+                  : null}
+              </div>
+            )}
+          </>
+        )}
+      </div>
```

Comment:
지금 코드는 삼항 연산자가 중첩되어 있어 코드 가독성에 좋지 않습니다.

아래처럼 겹치는 조건을 하나로 합쳐서 중첩 삼항연산자를 없앨 수도 있어요. 

이때 `result`는 검색 결과에 대한 배열이고, 기본값을 빈 배열로 두면 `null`을 쓰지 않아도 아무것도 없는 초기화면을 만들 수 있어요.

```suggestion
      <div className={styles.content}>
        {keyword && result.length === 0 ? (
          <div className={styles.noResult}>검색 결과가 없습니다</div>
        ) : (
          <div className={styles.posts}>
            {result.map((post) => (
              <PostBar key={post.postId} data={post} />
            ))}
          </div>
        )}
      </div>
```